bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -156,26 +161,110 @@ private void updateSchedulingBlockEnabledState(final boolean enabled) {
+            return new Some<>(((TableCoordinateSelection) selection).getCoordinates());
+        else
+            return None.instance();
     }
 
     /**
-     * Auxiliary method to determine if the current selection is the base position.
+     * Extract the base position out of the asterism.
+     * It could be a target or a sky position.
+     * Due to type inference issues with map, we have to do this in a horrible way.
      */
-    private boolean selectionIsBasePosition() {
+    private ImEither<SPTarget, SPCoordinates> basePosition() {
         final TargetEnvironment env = getDataObject().getTargetEnvironment();
-        return selectedTarget().exists((target) -> target == env.getArbitraryTargetFromAsterism());
+        final Asterism a            = env.getAsterism();
+
+        switch (a.asterismType()) {
+            case Single:
+                return new Left<>(((Asterism.Single) a).t());
+            case GhostSingleTarget:
+                final GhostAsterism.SingleTarget gsa = (GhostAsterism.SingleTarget) a;
+                if (gsa.base().isDefined()) return new Right<>(gsa.base().get());
+                else return new Left<>(gsa.target().spTarget());
+            case GhostDualTarget:
+                final GhostAsterism.DualTarget gda = (GhostAsterism.DualTarget) a;
+                if (gda.base().isDefined()) return new Right<>(gda.base().get());
+                else {
+                    final Option<Coordinates> cOpt = ImOption.fromScalaOpt(gda.basePosition(ImOption.scalaNone()));
+                    return new Right<>(cOpt.map(SPCoordinates::new).getOrElse(SPCoordinates::new));
+                }
+            case GhostTargetPlusSky:
+                final GhostAsterism.TargetPlusSky gtsa = (GhostAsterism.TargetPlusSky) a;
+                if (gtsa.base().isDefined()) return new Right<>(gtsa.base().get());
+                else return new Left<>(gtsa.target().spTarget());
+            case GhostSkyPlusTarget:
+                final GhostAsterism.SkyPlusTarget gsta = (GhostAsterism.SkyPlusTarget) a;
+                if (gsta.base().isDefined()) return new Right<>(gsta.base().get());
+                else return new Left<>(gsta.target().spTarget());
+            case GhostHighResolutionTarget:
+                final GhostAsterism.HighResolutionTarget ghta = (GhostAsterism.HighResolutionTarget) a;
+                if (ghta.base().isDefined()) return new Right<>(ghta.base().get());
+                else return new Left<>(ghta.target().spTarget());
+            case GhostHighResolutionTargetPlusSky:
+                final GhostAsterism.HighResolutionTargetPlusSky ghtsa = (GhostAsterism.HighResolutionTargetPlusSky) a;
+                if (ghtsa.base().isDefined()) return new Right<>(ghtsa.base().get());
+                else return new Left<>(ghtsa.target().spTarget());
+        }
+
+        // Make the compiler shut up.
+        return new Right<>(new SPCoordinates());
```

Comment:
I prefer to throw an exception when something unpossible happens.

Reply:
👍